test/Interop/Cxx/class/Inputs/constructors.h

```diff
@@ -176,4 +178,233 @@ class Value {
+} // namespace CtorWithDefaultArg
+
+void Retain11(CtorWithDefaultArg::CxxRefTy *_Nonnull v) {};
+void Release11(CtorWithDefaultArg::CxxRefTy *_Nonnull v) {};
+
+namespace CtorWithDefaultAndNonDefaultArg {
+struct __attribute__((swift_attr("import_reference")))
+__attribute__((swift_attr("retain:Retain12")))
+__attribute__((swift_attr("release:Release12"))) CxxRefTy {
+public:
+  int val;
+  CxxRefTy(int x, int y = 42) : val(x + y) {}
+};
+} // namespace CtorWithDefaultAndNonDefaultArg
+
+void Retain12(CtorWithDefaultAndNonDefaultArg::CxxRefTy *_Nonnull v) {};
+void Release12(CtorWithDefaultAndNonDefaultArg::CxxRefTy *_Nonnull v) {};
+
+namespace DefaulltAndNonDefaultCtors {
+struct __attribute__((swift_attr("import_reference")))
+__attribute__((swift_attr("retain:Retain13")))
+__attribute__((swift_attr("release:Release13"))) CxxRefTy {
+public:
+  int val = 1;
+  CxxRefTy() = default;
+  CxxRefTy(int x) : val(x) {}
+};
+} // namespace DefaulltAndNonDefaultCtors
+
+void Retain13(DefaulltAndNonDefaultCtors::CxxRefTy *_Nonnull v) {};
+void Release13(DefaulltAndNonDefaultCtors::CxxRefTy *_Nonnull v) {};
+
+namespace ParameterisedCtor {
+struct __attribute__((swift_attr("import_reference")))
+__attribute__((swift_attr("retain:Retain14")))
+__attribute__((swift_attr("release:Release14"))) CxxRefTy {
+public:
+  int val = 1;
+  CxxRefTy(int x) : val(x) {}
+};
+} // namespace ParameterisedCtor
+
+void Retain14(ParameterisedCtor::CxxRefTy *_Nonnull v) {};
+void Release14(ParameterisedCtor::CxxRefTy *_Nonnull v) {};
+
+namespace UserProvidedStaticFactory {
+struct __attribute__((swift_attr("import_reference")))
+__attribute__((swift_attr("retain:Retain15")))
+__attribute__((swift_attr("release:Release15"))) CxxRefTy {
+public:
+  int val = 1;
+
+  CxxRefTy() = default;
+
+  __attribute__((swift_name("init()")))
+  __attribute__((swift_attr("returns_retained")))
+  static CxxRefTy *_Nonnull create() {
+    CxxRefTy *_Nonnull returnPtr = new CxxRefTy();
+    returnPtr->val = 2;
+    return returnPtr;
+  }
+
+  __attribute__((swift_name("init(_:)")))
+  __attribute__((swift_attr("returns_retained")))
+  static CxxRefTy *_Nonnull create(int x) {
+    CxxRefTy *returnPtr = new CxxRefTy();
+    returnPtr->val = x;
+    return returnPtr;
+  }
+};
+} // namespace UserProvidedStaticFactory
+
+void Retain15(UserProvidedStaticFactory::CxxRefTy *_Nonnull v) {};
+void Release15(UserProvidedStaticFactory::CxxRefTy *_Nonnull v) {};
+
+// TODO: Avoid synthesizing factories for immortal references to prevent
+// "returns_(un)retained" errors and ensure "cannot be constructed" error
+// instead.
+namespace ImmortalReference {
+struct __attribute__((swift_attr("import_reference")))
+__attribute__((swift_attr("retain:immortal")))
+__attribute__((swift_attr("release:immortal")))
+CxxRefTy { //  expected-error {{'__returns_CxxRefTy' cannot be annotated with either SWIFT_RETURNS_RETAINED or SWIFT_RETURNS_UNRETAINED because it is not returning a SWIFT_SHARED_REFERENCE type}}
```

Comment:
We should not add this annotation in the generated code for immortal references.